src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -54,6 +47,33 @@ def register_options(cls, register):
             "library, invoked by a wrapper script.",
         )
 
+        interpreter_constraints_help = (
+            "Python interpreter constraints for this tool. An empty list uses the default "
+            "interpreter constraints for the repo."
+        )
+        if cls.register_interpreter_constraints:
+            register(
+                "--interpreter-constraints",
+                type=list,
+                advanced=True,
+                default=cls.default_interpreter_constraints,
+                help=interpreter_constraints_help,
+            )
+        else:
+            register(
+                "--interpreter-constraints",
+                type=list,
+                advanced=True,
+                default=[],
+                help=interpreter_constraints_help,
+                removal_version="2.1.0.dev0",
+                removal_hint=(
+                    "This option no longer does anything, as Pants auto-configures the interpreter "
+                    f"constraints for {cls.options_scope} based on your code's interpreter "
+                    "constraints."
+                ),
+            )
```

Comment:
So no need to do this, but one alternative is to build up the kwargs separately, then do a single call to register. I think we used to do this for BinaryTool options, for example. e.g.:
```python
if cls.register_interpreter_constraints:
    interpreter_constraints_kwargs = dict(default=cls.default_interpreter_constraints)
else:
    interpreter_constraints_kwargs = dict(
        default=[],
        removal_version="2.1.0.dev0",
        removal_hint="...",
    )
register("--interpreter-constraints", type=list, advanced=True, help="...",
         **interpreter_constraints_kwargs)
```

Reply:
Oh, true. This code gets deleted in 2.1.0 though, so I think it's fine as is.

Reply:
Ok great!